dlp/snippets/inspect/inspect_bigquery_with_sampling.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package inspect
+
+// [START dlp_inspect_bigquery_with_sampling]
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	dlp "cloud.google.com/go/dlp/apiv2"
+	"cloud.google.com/go/dlp/apiv2/dlppb"
+	"cloud.google.com/go/pubsub"
+)
+
+// inspectBigQueryTableWithSampling inspect bigQueries for sensitive data with sampling
+func inspectBigQueryTableWithSampling(w io.Writer, projectID, topicID, subscriptionID string) error {
+	// projectId := "your-project-id"
+	// topicID := "your-pubsub-topic-id" or provide a topicID name to create one
+	// subscriptionID := "your-pubsub-topic-id" or provide a topicID name to create one
+
+	ctx := context.Background()
+
+	// Initialize a client once and reuse it to send multiple requests. Clients
+	// are safe to use across goroutines. When the client is no longer needed,
+	// call the Close method to cleanup its resources.
+	client, err := dlp.NewClient(ctx)
+	if err != nil {
+		return err
```

Comment:
it's good practice to use fmt.Errorf to wrap the error and give a message at least indicating the source:
`fmt.Errorf("dlp.NewClient: %w", err)`

Same for other errors below.

Reply:
- Waiting for your reply on this [comment](https://github.com/GoogleCloudPlatform/golang-samples/pull/3166#discussion_r1237975723)

Reply:
Discussed with @telpirion , you can leave this as is. Sorry for the randomization.